### PR TITLE
SNO-138-fix-pytest-warnings

### PR DESCRIPTION
**Pull request by paul-sud**

### CODE REVIEW

File: src/snovault/elasticsearch/searches/queries.py [hunk 1]
```diff
@@ -198,13 +198,13 @@ def _get_collection_names_for_item_types(self, item_types):
...
-        return query.replace('/', '\/')
+        return query.replace('/', '\\/')
 
     def _escape_fuzzy_tilde(self, query):
-        return query.replace('~', '\~')
+        return query.replace('~', '\\~')
 
     def _escape_boost_caret(self, query):
-        return query.replace('^', '\^')
+        return query.replace('^', '\\^')
```
Comment (keenangraham) [inline src/snovault/elasticsearch/searches/queries.py:207]: Which warnings does this fix? I'm a little worried we are actually changing the meaning of the query by adding additional backslash.

https://www.elastic.co/guide/en/elasticsearch/reference/5.6/query-dsl-query-string-query.html#_reserved_characters

https://lucene.apache.org/core/3_4_0/queryparsersyntax.html#Escaping%20Special%20Characters

These specific cases are to nullify characters in user input that could cause expensive queries/timeouts. By escaping the backslash itself doesn't it become the literal character interpreted by ES, while the `~` or `^` is again treated as a special character?

**Review by keenangraham — CHANGES_REQUESTED** [review]

**Comment by keenangraham** [comment]
I think maybe there's an issue with adding additional backslash to query string builder. The rest looks fine.

File: src/snovault/elasticsearch/searches/queries.py (hunk 1 above)
Comment (paul-sud) [reply src/snovault/elasticsearch/searches/queries.py:207]: You get an `invalid escape sequence` warning. Backslash is a reserved character in Python strings that should be escaped (unless you use raw string). The resulting strings with the change here are identical.

```
>>> "~".replace('~', '\~') == "~".replace('~', '\\~')
True
>>> "/".replace('/', '\/') == "/".replace('/', '\\/')
True
>>> "^".replace('^', '\^') == "^".replace('^', '\\^')
True
```

Comment (keenangraham) [reply src/snovault/elasticsearch/searches/queries.py:207]: Ah, makes sense. Thanks for fixing.

**Review by keenangraham — APPROVED** [review]

**Pull request merged**